diff --git a/third_party/sqlite-vec/sqlite-vec.c b/third_party/sqlite-vec/sqlite-vec.c
index 7af3b6a7..1ab28bde 100644
--- a/third_party/sqlite-vec/sqlite-vec.c
+++ b/third_party/sqlite-vec/sqlite-vec.c
@@ -3763,6 +3763,8 @@ void vec0_free(vec0_vtab *p) {
   }
 
   for (int i = 0; i < p->numMetadataColumns; i++) {
+    sqlite3_free(p->shadowMetadataChunksNames[i]);
+    p->shadowMetadataChunksNames[i] = NULL;
     sqlite3_free(p->metadata_columns[i].name);
     p->metadata_columns[i].name = NULL;
   }
